WordPress/Classes/ViewRelated/Plugins/CollectionViewContainerRow.swift

```diff
@@ -0,0 +1,178 @@
+/// A `ImmuTableRow` that contains a `UICollectionView`.
+/// Currently the `CollectionViewContainerCell` is fairly specific to needs of Plugin Directory,
+/// but it should be possible to make it more generic in the future if there's a need to — it was deliberately
+/// not made as generic as possible, in case it won't be used.
+
+private let CellReuseIdentifier = "CollectionViewContainerRowReuseIdentifier"
+
+struct CollectionViewContainerRow<CollectionViewCellType: UICollectionViewCell, Item>: ImmuTableRow {
+    typealias CellType = CollectionViewContainerCell
+
+    static var cell: ImmuTableCell {
+        return .class(CollectionViewContainerCell.self)
+    }
+
+    let title: String
+    let secondaryTitle: String?
+    let action: ImmuTableAction?
+
+    private let helper: CollectionViewContainerRowHelper
+
+    init(data: [Item],
+         title: String,
+         secondaryTitle: String?,
+         action: ImmuTableAction?,
+         configureCollectionCell: @escaping ((CollectionViewCellType, Item) -> Void),
+         collectionCellSelected: @escaping ((Item) -> Void)) {
+
+        self.title = title
+        self.secondaryTitle = secondaryTitle
+        self.action = action
+
+        let configurationBlock = { (cell: UICollectionViewCell, item: Any) in
+            configureCollectionCell(cell as! CollectionViewCellType, item as! Item)
+        }
+
+        let selectionBlock = { (item: Any) in
+            collectionCellSelected(item as! Item)
+        }
+
+        helper = CollectionViewContainerRowHelper(dataSourceItems: data,
+                                                  configureCollectionCellBlock: configurationBlock,
+                                                  collectionCellSeletectedBlock: selectionBlock)
+    }
+
+
+    func configureCell(_ cell: UITableViewCell) {
+        let cell = cell as! CellType
+
+        if nil != Bundle.main.path(forResource: String(describing: CollectionViewCellType.self), ofType: "nib") {
+            cell.collectionView.register(UINib(nibName: String(describing: CollectionViewCellType.self), bundle: nil),
+                                         forCellWithReuseIdentifier: CellReuseIdentifier)
+        } else {
+            cell.collectionView.register(CollectionViewCellType.self, forCellWithReuseIdentifier: CellReuseIdentifier)
+        }
+
+        cell.titleLabel.text = title
+
+        cell.actionButton.setTitle(secondaryTitle, for: .normal)
+        cell.buttonTappedAction = { self.action?(self) }
+
+        cell.collectionView.delegate = helper
+        cell.collectionView.dataSource = helper
+    }
+
+
+}
+
+/// Generic Swift classes can't be made `@objc`, so we're using this helper object to be a `UICollectionView[DataSource|Delegate]`
+@objc private class CollectionViewContainerRowHelper: NSObject, UICollectionViewDelegate, UICollectionViewDataSource {
```

Comment:
Since this internal class deals with a UICollectionViewDelegate + Datasource, and it doesn't really have knowledge of *where* it's being used, it could probably use a more explicit name. How about `CollectionViewHandler`?

Reply:
Excellent suggestion, thanks!